### ISSUE TITLE
Only one profile for efficiency possible

### ISSUE THREAD

**Issue by MaGering** [oemof/oemof-tabular#117]
It is not possible to pass different foreign efficiency profiles, since the sequences name written into the `datapackage.json` is `efficiency_profile` for all components that have `efficiency` as attribute.

This is due to [this code](https://github.com/oemof/oemof-tabular/blob/6476a18e4be117627d873afbf71f2895f1ae4e1f/src/oemof/tabular/datapackage/building.py#L134-L146):

```
            # Define all undefined foreign keys for as <var name>_profile
            for key in foreign_keys:
                if key not in (
                    ["profile"] + list(config.FOREIGN_KEY_DESCRIPTORS)
                ):
                    if r.name in foreign_keys[key]:
                        r.descriptor["schema"]["foreignKeys"].append(
                            {
                                "fields": key,
                                "reference": {"resource": key + "_profile"},
                            }
                        )

```

In case where `key` equals to `efficiency` the name is set with `reference` to `efficiency_profile`. Instead it should be set to `r.name + "_profile"`

**Comment by nailend** [comment]
@MaGering I am not sure if I understand the problem. Wouldn't it be enough to just pass another foreign key to the `infer` method  and save the additional profiles in a separate csv file? At least, this is how its currently handled.

**Comment by nailend** [comment]
I think I am getting it: You are having multiple components of the same type, e.g. heat_pumps, but you don't want to give all of them the same efficiencies due to e.g. local differences. 

In this case it should be enough to write different names in the cells of the `efficiency` column of your component. 
Compare with the [volatile component](https://github.com/oemof/oemof-tabular/blob/dev/src/oemof/tabular/examples/datapackages/dispatch/data/elements/volatile.csv).

**Comment by MaGering** [comment]
> I think I am getting it: You are having multiple components of the same type, e.g. heat_pumps, but you don't want to give all of them the same efficiencies due to e.g. local differences.
> 
> In this case it should be enough to write different names in the cells of the `efficiency` column of your component. Compare with the [volatile component](https://github.com/oemof/oemof-tabular/blob/dev/src/oemof/tabular/examples/datapackages/dispatch/data/elements/volatile.csv) and respective [volatile_profiles](https://github.com/oemof/oemof-tabular/blob/dev/src/oemof/tabular/examples/datapackages/dispatch/data/sequences/volatile_profile.csv)

Yes exactly, that's the case where the problem occurs. For the field `profile` this works just fine. But how about that efficiency or marginal_cost etc.?

Concerning the solution in your first comment: Will oemof.tabular map it to the correct attribute if I put for example "heatpump_small_efficiency" in front of it? I assumed that the key must be named exactly the same as the name of the attribute. So that the key must be called `marginal_cost` for example and not `wind_marginal_cost`.

**Comment by iuljan** [comment]
@MaGering sorry for not answering... is this solved anyways?
